Sources/SwiftSyntax/Trivia+commentValue.swift

```diff
@@ -0,0 +1,143 @@
+      var lines = text.dropPrefix(isDocComment ? "/**" : "/*").dropSuffix("*/")
+        .split(omittingEmptySubsequences: false, whereSeparator: \.isNewline)
+
+      // If the comment content starts on the same line as the `/*` marker or ends on the same line as the `*/` marker,
+      // it is common to separate the marker and the actual comment using spaces. Strip those spaces if they exists.
+      // If there are non no-space characters on the first / last line, then the comment doesn't start / end on the line
+      // with the marker, so don't do the stripping.
+      if let firstLine = lines.first, firstLine.contains(where: { $0 != " " }) {
+        lines[0] = firstLine.drop { $0 == " " }
+      }
+      if let lastLine = lines.last, lastLine.contains(where: { $0 != " " }) {
+        lines[lines.count - 1] = lastLine.dropLast { $0 == " " }
+      }
+
+      var indentation: Substring? = nil
+      // Find the lowest indentation that is common among all lines in the block comment. Do not consider the first line
+      // because it won't have any indentation since it starts with /*
+      for line in lines.dropFirst() {
+        let lineIndentation = line.prefix(while: { $0 == " " || $0 == "\t" })
+        guard let previousIndentation = indentation else {
+          indentation = lineIndentation
+          continue
+        }
+        indentation = commonPrefix(previousIndentation, lineIndentation)
+      }
+
+      guard let firstLine = lines.first else {
+        // We did not have any lines. This should never happen in practice because `split` never returns an empty array
+        // but be safe and return `nil` here anyway.
+        return nil
+      }
+
+      var unindentedLines = [firstLine] + lines.dropFirst().map { $0.dropPrefix(indentation ?? "") }
+
+      // If the first line only contained the comment marker, don't include it. We don't want to start the comment value
+      // with a newline if `/*` is on its own line. Same for the end marker.
+      if unindentedLines.first?.allSatisfy({ $0 == " " }) ?? false {
+        unindentedLines.removeFirst()
+      }
+      if unindentedLines.last?.allSatisfy({ $0 == " " }) ?? false {
+        unindentedLines.removeLast()
+      }
+      // We canonicalize the line endings to `\n` here. This matches how we concatenate the different line comment
+      // pieces using \n as well.
+      return unindentedLines.joined(separator: "\n")
+    }
+
+    for piece in pieces {
+      switch piece {
+      case .blockComment(let text), .docBlockComment(let text):
+        if hasBlockComment || hasLineComment {
+          return nil
+        }
+        hasBlockComment = true
+        guard let processedText = processBlockComment(text, isDocComment: piece.isDocComment) else {
+          return nil
+        }
+        comments.append(processedText[...])
+      case .lineComment(let text), .docLineComment(let text):
+        if hasBlockComment {
```

Comment:
So we allow
```swift
/// Foo Bar
/// This is a doc-comment.
// @_spi(Testing)
public fooBar() {}
```
resulting `"Foo Bar\nThis is a doc-comment\n@_spi(Testing)"`, but not
```swift
/**
 * Foo Bar
 * This is a doc-comment.
 */
// @_spi(Testing)
public fooBar() {}
```

Not sure about this case.